src/hotkey/queue.c

```diff
@@ -0,0 +1,190 @@
+#include "queue.h"
+#include "constant.h"
+
+#include <cc_debug.h>
+#include <cc_mm.h>
+#include <cc_pool.h>
+
+#define QUEUE_MODULE_NAME "hotkey::queue"
+
+static bool queue_init = false;
+static int queue_size = 0;
+
+FREEPOOL(qn_pool, qnq, queue_node);
+static struct qn_pool qnp;
+static bool qnp_init = false;
+
+struct queue_node {
+    char                     key[MAX_KEY_LEN];
```

Comment:
I will do this in a separate pull request. This is a larger change than it seems, as in order to do this we need to make the queue aware of the counting map